first_assesment/repository/pokemon_repository.go

```diff
@@ -0,0 +1,56 @@
+package repository
+
+import (
+	"errors"
+	"os"
+
+	"first/model"
+
+	"github.com/gocarina/gocsv"
+)
+
+type PokemonRepository struct {
+	file string
+}
+
+func NewPokemonRepository() (*PokemonRepository, error) {
+	return &PokemonRepository{
+		file: "csv/input_file.csv",
+	}, nil
+
+}
+
+func (p *PokemonRepository) GetAll() ([]*model.Pokemon, error) {
```

Comment:
Just curiosity, why are you returning an array of model.Pokemon pointers here instead of an array of model.Pokemon?

Reply:
because my method try to return 2 values, the array and the error, the problem is that if it is a simple array I cannot return something like that:

return nil,err

because when I try to execute the code I receive the following message:

repository/pokemon_repository.go:48:3: cannot use nil as type model.Pokemon in return argument

Do you know another way to solve that?

Reply:
Yes, you can return `nil` with either `[]model.Pokemon` or `[]*model.Pokemon`.

https://play.golang.org/p/wlxCisZW7cv

I asked that because normally when you use pointers, is because you are going to modify the variables by reference and not by value, but in this case you are not doing that.

And what you mention. in line 48, well, that's another function, and that function is okay to return a `*model.Pokemon`. Now, there's a difference between what `GetAll()` and `GetByID` returns. Instead of returning `pokemon` in `GetByID` you can return `&pokemon` (the pointer address of pokemon variable).

Reply:
fixed it, thanks